### PR TITLE
Less verbose HLTPrescaleProvider (81X)

### PR DESCRIPTION
**Pull request by Martin-Grunewald**
Less verbose HLTPrescaleProvider (81X)

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
please test

File: HLTrigger/HLTcore/interface/HLTPrescaleProvider.h [hunk 1]
```diff
@@ -78,6 +78,8 @@ class HLTPrescaleProvider {
 
   HLTConfigProvider hltConfigProvider_;
   L1GtUtils l1GtUtils_;
+  unsigned int count_[5] = {0,0,0,0,0};
```
Comment (Dr15Jones) [inline HLTrigger/HLTcore/interface/HLTPrescaleProvider.h]: Seems like it could be an unsigned char

**Comment by fwyzard** [comment]
Instead of hardcoding the limits in the code, is it possible to achieve the same effect of limiting the occurrence of the different messages with some settings of the MessageLogger ?
Though, I guess that would require using a different category for each message ?

**Comment by Martin-Grunewald** [comment]
I do not have time to go through CMSSW and find all ML configs and then change those.
Let me rephrase that.: I do not want to spend time on going through CMSSW and find all ML configs and then change those.

**Comment by fwyzard** [comment]
@Martin-Grunewald , I wouldn't ask you to do such thing...

For production workflows, I assume there is a central configuration for the MessageLogger. If there is a way to achieve a similar result, I'd rather configure the MessageLogger accordingly, than hardcode limits on the number of messages in the code.

Otherwise this defeats the purpose of using the MessageLogger in the first place.

**Comment by Martin-Grunewald** [comment]
@davidlange6 should know...

**Comment by davidlange6** [comment]
I don’t - maybe @Dr15Jones can help.

**Comment by Dr15Jones** [comment]
I don't know if we use a central message logger configuration for production. I guess we'd have to look at a configuration to determine that. I also do not know if there is a reasonable way to limit the number of outputs. I fear that if we do, we also shut down the 'starting event #' messages as well.
Another problem is I think filtering with the message logger happens only after the message has been fully formatted which takes time and it would happen after adding to its internal buffer if multiple threads are writing very quickly.

**Comment by Dr15Jones** [comment]
The point I was trying to make is it is much better to never send the message in the first place instead of trying to make the message logger filter. I also agree with Andrea that such a statement does defeat one of the purposes of the MessageLogger. But the code is as it is.

**Comment by Dr15Jones** [comment]
So I dug up the MessageLogger documentation. There is no way to set a true 'limit'. The actual 'limit' just sets an exponential back-off
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLoggerParameters#Message_Logging_Concepts

With that in mind, the documentation does give an example
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLoggerParameters#Examples_of_Limits_and_Timespan

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1
